agent/src/main/java/com/code_intelligence/jazzer/instrumentor/HookMethodVisitor.kt

```diff
@@ -292,10 +264,18 @@ private class HookMethodVisitor(
         Opcodes.INVOKESPECIAL
     )
 
-    private fun findMatchingHook(hookType: HookType, owner: String, name: String, descriptor: String): Hook? {
-        val withoutDescriptorKey = "$hookType#$owner#$name"
-        val withDescriptorKey = "$withoutDescriptorKey#$descriptor"
-        return hooks[withDescriptorKey] ?: hooks[withoutDescriptorKey]
+    private fun findMatchingHooks(owner: String, name: String, descriptor: String): List<Hook> {
+        val result = HookType.values().map { hookType ->
+            val withoutDescriptorKey = "$hookType#$owner#$name"
+            val withDescriptorKey = "$withoutDescriptorKey#$descriptor"
+            hooks[withDescriptorKey].orEmpty() + hooks[withoutDescriptorKey].orEmpty()
+        }.flatten().sortedBy { it.hookType }
+        require(
```

Comment:
IMHO using `check` would convey the situation a little bit better, because it throws an `IllegalStateException` and not an `IllegalArgumentException` like `require`. Providing an error message as second argument, describing that only one `HookType.REPLACE` on the given method is allowed, could also be helpful.